src/content/en/fundamentals/performance/webpack/decrease-frontend-size.md

```diff
@@ -0,0 +1,691 @@
+/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__comments_css__ = __webpack_require__(1);
+/* harmony import */ var __WEBPACK_IMPORTED_MODULE_0__comments_css_js___default =
+__webpack_require__.n(__WEBPACK_IMPORTED_MODULE_0__comments_css__);
+
+function render(data, target) {
+  console.log('Rendered!');
+}
+</pre>
+
+</li>
+
+<li>
+
+The UglifyJS plugin minifies it into approximately the following:
+
+<pre class="prettyprint">
+// minified bundle.js (part of)
+"use strict";function t(e,n){console.log("Rendered!")}
+Object.defineProperty(n,"__esModule",{value:!0}),n.render=t;var o=r(1);r.n(o)
+</pre>
+
+</li>
+</ol>
+
+The plugin comes bundled with webpack. To enable it, add it to the `plugins` section of the config:
+
+    // webpack.config.js
+    const webpack = require('webpack');
+
+    module.exports = {
+      plugins: [
+        new webpack.optimize.UglifyJsPlugin(),
+      ],
+    };
+
+The second way is loader-specific options ([what a loader
+is](https://webpack.js.org/concepts/loaders/)). With loader options, you can compress things that
+the UglifyJS plugin can’t minify. For example, when you import a CSS file with
+[`css-loader`](https://github.com/webpack-contrib/css-loader), the file is compiled into a string:
+
+    /* comments.css */
+    .comment {
+      color: black;
+    }
+
+↓
+
+<pre class="prettyprint">
+// minified bundle.js (part of)
+exports=module.exports=__webpack_require__(1)(),
+exports.push([module.i,<strong>".comment {\r\n  color: black;\r\n}"</strong>,""]);
+</pre>
+
+UglifyJS can’t compress this code because it’s a string. To minify the file content, we need to
+configure the loader to do this:
+
+<pre class="prettyprint">
+// webpack.config.js
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.css$/,
+        use: [
+          'style-loader',
+          { loader: 'css-loader', options: <strong>{ minimize: true }</strong> },
+        ],
+      },
+    ],
+  },
+};
+</pre>
+
+Note: The UglifyJS plugin can’t compile the ES2015+ (ES6+) code. This means that if your code uses
+classes, arrow functions or other new language features, and you don’t compile them into ES5, the
+plugin will throw an error. <br><br> If you need to compile the new syntax, use the
+[uglifyjs-webpack-plugin](https://github.com/webpack-contrib/uglifyjs-webpack-plugin) package. This
+is the same plugin that’s bundled with webpack, but newer, and it’s able to compile the ES2015+
+code.
+
+### Further reading {: .hide-from-toc }
+
+* [The UglifyJsPlugin docs](https://github.com/webpack-contrib/uglifyjs-webpack-plugin)
+
+* Other popular minifiers: [Babel
+  Minify](https://github.com/webpack-contrib/babel-minify-webpack-plugin), [Google Closure
+  Compiler](https://github.com/roman01la/webpack-closure-compiler)
+
+## Specify `NODE_ENV=production`
+
+Another way to decrease the front-end size is to set the `NODE_ENV`
+[environmental variable](https://superuser.com/questions/284342/what-are-path-
```

Comment:
don't linebreak URLs, otherwise the link breaks. it's OK if this exceeds 80 chars